app/controllers/campaigns_controller.rb

```diff
@@ -51,7 +54,15 @@ def create
     Time.use_zone(current_nonprofit.timezone || 'UTC') do
       params[:campaign][:end_datetime] = Chronic.parse(params[:campaign][:end_datetime]) if params[:campaign][:end_datetime].present?
     end
-    campaign = current_nonprofit.campaigns.create params[:campaign]
+
+    if !params[:campaign][:parent_campaign_id]
```

Comment:
Are there checks for situations where the parent_campaign_id is invalid or doesn't belong to the nonprofit the new campaign is being created for?

Reply:
The nonprofit_id is deduced from the parent_campaign_id. I added a check for when the parent_campaign_id is invalid.